### PR TITLE
Update release notes for oneDPL 2021.5 release

### PR DESCRIPTION
**Pull request by ValentinaKats**
Signed-off-by: valentina.kats@intel.com

### CODE REVIEW

File: documentation/release_notes.rst [hunk 1]
```diff
@@ -8,6 +8,61 @@ The Intel® oneAPI DPC++ Library (oneDPL) accompanies the Intel® oneAPI DPC++/C
...
+New in 2021.5
+=============
+
+New Features
+------------
+- Added new random number distributions: ``exponential_distribution``, ``bernoulli_distribution``,
+  ``geometric_distribution``, ``lognormal_distribution``, ``weibull_distribution``, ``cachy_distribution``, ``extreme_value_distribution``.
+- Added the serial-based versions of the following algorithms: ``all_of``, ``any_of``, 
+  ``none_of``, ``count``, ``count_if``, ``for_each``, ``find``, ``find_if``, ``find_if_not``.
+  For the detailed list, please refer to `Tested Standard C++ API Reference`_.
```
Comment (haonanya) [inline documentation/release_notes.rst:20]: LGTM aside from a couple nits. 
I am not able arrive at correct position. Maybe use `Tested Standard C++ API Reference <https://github.com/oneapi-src/oneDPL/blob/main/documentation/library_guide/tested_standard_cpp_api.rst#tested-standard-c-api-reference>`_ instead.
Please see https://github.com/haonanya/oneDPL/commit/a33b9ddee9fed74baf6dab463dd8edeb5d3b03d3 for details.
If there is something wrong, please correct me.
Thanks very much.

File: documentation/release_notes.rst [hunk 2]
```diff
@@ -8,6 +8,61 @@ The Intel® oneAPI DPC++ Library (oneDPL) accompanies the Intel® oneAPI DPC++/C
...
+- Fixed hanging of reduction-based algorithms (``count``, ``count_if``, ``is_partitioned``, ``lexicographical_compare``, ``max_element``, ``min_element``, ``minmax_element``,    ``reduce``, ``transform_reduce``) used with DPC++ policies. The hanging occurred when the algorithms handled elements with a large size (e.g. 64 bytes and more) and a program used a lot of oneDPL algorithms and DPC++ kernels.
+- Fixed a possible data race for the following algorithms with DPC++ policies: ``count``, ``count_if``, ``is_partitioned``,
+  ``lexicographical_compare``, ``max_element``, ``min_element``, ``minmax_element``, ``transform_reduce``. 
+
+Known Issues and Limitations
+----------------------------
+
+New in This Release
+^^^^^^^^^^^^^^^^^^^
+- ``ranlux_24_48_test`` test may hang on Ubuntu* OS.
```
Comment (paveldyakov) [inline documentation/release_notes.rst:37]: It is not reproduced now, could you please remove it?

File: documentation/release_notes.rst (hunk 1 above)
Comment (jiejanezhang) [reply documentation/release_notes.rst:20]: Link is updated to https://github.com/oneapi-src/oneDPL/blob/release/2021.5/documentation/library_guide/tested_standard_cpp_api.rst#tested-standard-c-api-reference%60_..

File: documentation/release_notes.rst (hunk 2 above)
Comment (jiejanezhang) [reply documentation/release_notes.rst]: Removed.

File: documentation/release_notes.rst [hunk 3]
```diff
@@ -8,6 +8,61 @@ The Intel® oneAPI DPC++ Library (oneDPL) accompanies the Intel® oneAPI DPC++/C
...
+- Improved performance of ``search`` and ``find_end`` algorithms on GPU devices.
+
+Fixed Issues
+------------
+- Fixed SYCL* 2020 features deprecation warnings.
+- Fixed some corner cases of ``normal_distribution`` functionality.
+- Fixed a floating point exception occurring on CPU devices when a program uses a lot of oneDPL algorithms and DPC++ kernels.
+- Fixed hanging of reduction-based algorithms (``count``, ``count_if``, ``is_partitioned``, ``lexicographical_compare``, ``max_element``, ``min_element``, ``minmax_element``,    ``reduce``, ``transform_reduce``) used with DPC++ policies. The hanging occurred when the algorithms handled elements with a large size (e.g. 64 bytes and more) and a program used a lot of oneDPL algorithms and DPC++ kernels.
+- Fixed a possible data race for the following algorithms with DPC++ policies: ``count``, ``count_if``, ``is_partitioned``,
+  ``lexicographical_compare``, ``max_element``, ``min_element``, ``minmax_element``, ``transform_reduce``.
```
Comment (dsobolev-dev) [inline documentation/release_notes.rst:30]: Probably we can unite these bullets omitting extra information in the first one.

- Fixed possible hanging and data races of the following algorithms used with DPC++ execution policies: ``count``, ``count_if``, ``is_partitioned``, ``lexicographical_compare``, ``max_element``, ``min_element``, ``minmax_element``, ``transform_reduce``.

Comment (jiejanezhang) [reply documentation/release_notes.rst]: > Probably we can unite these bullets omitting extra information in the first one.
> 
> * Fixed possible hanging and data races of the following algorithms used with DPC++ execution policies: `count`, `count_if`, `is_partitioned`, `lexicographical_compare`, `max_element`, `min_element`, `minmax_element`, `transform_reduce`.

Merged the two bullets.

**Review by jiejanezhang — APPROVED** [review]

**Pull request merged**